workflow/funcs.go

```diff
@@ -0,0 +1,30 @@
+package workflow
+
+import (
+	"fmt"
+	"strings"
+)
+
+// templateFuncs defines the custom functions available to workflow templates.
+var templateFuncs = map[string]interface{}{
+	"contains":        strings.Contains,
+	"hasPrefix":       strings.HasPrefix,
+	"hasSuffix":       strings.HasSuffix,
+	"formatPartition": formatPartition,
+}
+
+// formatPartition formats a device path with partition for the device type. If it receives an
+// unidentifiable device path it returns the dev.
+//
+// Examples
+// 		formatPartition("/dev/nvme0n1", 0) -> /dev/nvme0n1p1
+// 		formatPartition("/dev/sda", 1) -> /dev/sda1
+func formatPartition(dev string, partition int) string {
```

Comment:
Are we OK with this name? I originally had `formatDevicePartition` and @displague called out its specific to Linux.